### PR TITLE
Remove deprecations from autoconfigure module

### PR DESCRIPTION
**Pull request by jack-berg**
The autoconfiguration module is looked pretty solid now with only `AutoConfiguredOpenTelemetrySdk` and its builder left as publicly accessible.

### CODE REVIEW

File: sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/OpenTelemetrySdkAutoConfiguration.java [hunk 1]
```diff
@@ -1,168 +0,0 @@
...
-  @Deprecated
-  public static OpenTelemetrySdk initialize(boolean setResultAsGlobal, ConfigProperties config) {
-    return AutoConfiguredOpenTelemetrySdk.builder()
-        .setResultAsGlobal(setResultAsGlobal)
-        .setConfig(config)
-        .build()
-        .getOpenTelemetrySdk();
-  }
-
-  static OpenTelemetrySdk newOpenTelemetrySdk(
```
Comment (jack-berg) [inline sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/OpenTelemetrySdkAutoConfiguration.java:87]: This logic was moved to `AutoConfiguredOpenTelemetrySdk#build()`, and `#configureMeterProvider` was moved to `MeterProviderConfiguration`.

File: sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/ResourceConfiguration.java [hunk 2]
```diff
@@ -5,48 +5,24 @@
...
 
-/**
- * Auto-configuration for the OpenTelemetry {@link Resource}.
- *
- * @deprecated Use {@link AutoConfiguredOpenTelemetrySdk#getResource()}.
- */
-@Deprecated
-public final class OpenTelemetryResourceAutoConfiguration {
+/** Auto-configuration for the OpenTelemetry {@link Resource}. */
+final class ResourceConfiguration {
```
Comment (jack-berg) [inline sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/ResourceConfiguration.java:20]: Renamed and merged with `EnvironmentResource`.

**Comment by jkwatson** [comment]
I want to hold off on this for a moment. @breedx-splk just found a use-case that we may have broken with these removals, namely being able to just get the auto-configured Resource, without building a whole SDK.

**Review by breedx-splk — COMMENTED** [review]
Hey @jack-berg this is good stuff. Coincidentally, I just filed #3924 around the same time due to there not being a clear approach to handling the Resource from agent code. 

I would prefer that we get #3924 figured out prior to removing all these deprecations (or if we come up with a simple idea, as part of this PR maybe).

**Review by jkwatson — APPROVED** [review]

**Comment by jack-berg** [comment]
Resolved conflicts. Should be good to go now 👍

**Review by anuraaga — APPROVED** [review]

**Pull request merged**